x-pack/plugins/security_solution/public/management/components/endpoint_responder/get_running_processes_action.tsx

```diff
@@ -0,0 +1,169 @@
+    },
+    EndpointCommandDefinitionMeta
+  >
+>(({ command, setStore, store, status, setStatus, ResultComponent }) => {
+  const endpointId = command.commandDefinition?.meta?.endpointId;
+  const { actionId, completedActionDetails } = store;
+
+  const isPending = status === 'pending';
+  const actionRequestSent = Boolean(store.actionRequestSent);
+
+  const getRunningProcessesApi = useSendGetEndpointRunningProcessesRequest();
+
+  const { data: actionDetails } = useGetActionDetails<RunningProcessesEntry>(actionId ?? '-', {
+    enabled: Boolean(actionId) && isPending,
+    refetchInterval: isPending ? 3000 : false,
+  });
+
+  // Send get running processes request if not yet done
+  useEffect(() => {
+    if (!actionRequestSent && endpointId) {
+      getRunningProcessesApi.mutate({
+        endpoint_ids: [endpointId],
+        comment: command.args.args?.comment?.[0],
+      });
+
+      setStore((prevState) => {
+        return { ...prevState, actionRequestSent: true };
+      });
+    }
+  }, [actionRequestSent, command.args.args?.comment, endpointId, getRunningProcessesApi, setStore]);
+
+  // If get running processes request was created, store the action id if necessary
+  useEffect(() => {
+    if (getRunningProcessesApi.isSuccess && actionId !== getRunningProcessesApi.data.data.id) {
+      setStore((prevState) => {
+        return { ...prevState, actionId: getRunningProcessesApi.data.data.id };
+      });
+    }
+  }, [actionId, getRunningProcessesApi?.data?.data.id, getRunningProcessesApi.isSuccess, setStore]);
+
+  useEffect(() => {
+    if (actionDetails?.data.isCompleted) {
+      setStatus('success');
+      setStore((prevState) => {
+        return {
+          ...prevState,
+          completedActionDetails: actionDetails?.data,
+        };
+      });
+    }
+  }, [actionDetails?.data, setStatus, setStore]);
+
+  const tableEntries = useMemo(() => {
+    if (endpointId) {
+      return completedActionDetails?.outputs?.[endpointId]?.content.entries ?? [];
+    }
+    return [];
+  }, [completedActionDetails?.outputs, endpointId]);
+
+  // Show nothing if still pending
+  if (isPending) {
+    return <ResultComponent showAs="pending" showTitle={false} />;
+  }
+
+  // Show errors
+  if (completedActionDetails?.errors) {
+    return (
+      <ResultComponent
+        showAs="failure"
+        title={i18n.translate(
+          'xpack.securitySolution.endpointResponseActions.getRunningProcesses.errorMessageTitle',
+          { defaultMessage: 'Get running processes action failed' }
+        )}
+        data-test-subj="getRunningProcessesErrorCallout"
+      >
+        <FormattedMessage
+          id="xpack.securitySolution.endpointResponseActions.getRunningProcesses.errorMessage"
+          defaultMessage="The following errors were encountered: {errors}"
+          values={{ errors: completedActionDetails.errors.join(' | ') }}
+        />
+      </ResultComponent>
+    );
+  }
+
+  const columns = [
```

Comment:
suggestion to pull this up above line 108 and memoize the array with `useMemo()`